workload/schemachange: add INSPECT operation to random schema workload

spilchen opened the pull request:
Adds a new inspect operation to the schema change workload, enabling random generation of INSPECT TABLE and INSPECT DATABASE statements.

Features:
- Support for TABLE/DB targets, AS OF SYSTEM TIME
- Always runs in DETACHED mode so that it can be run inside a transaciton
- Results checked post-run via SHOW INSPECT ERRORS

Errors reported in JSON, consistent with existing workload logs

Closes #155483
Epic: CRDB-55075
Release note: none

fqazi reviewed (changes requested):
A few minor comments, nice work!

> @fqazi reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-:-Ocg3FnGA_Ui4XMFcVyp:b-b4k7x0)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @shailendra-patel and @williamchoe3)

___
*[`pkg/workload/schemachange/schemachange.go` line 210 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-Ocg1_ERCUTVQwDfeNg-:-Ocg1_ERCUTVQwDfeNg0:bmgky9s) ([raw file](https://github.com/cockroachdb/cockroach/blob/fafc7438b6b7a6969bc904f995fc63a1d87828f6/pkg/workload/schemachange/schemachange.go#L210)):*
> ```Go
> 	var totalErrors int
> 	for _, jobID := range jobIDs {
> 		query := fmt.Sprintf("SHOW INSPECT ERRORS FOR JOB %d WITH DETAILS", jobID)
> ```

Nit: Do we want to explicit select columns via `SELECT <COLUMNS> FROM [SHOW ...]`  for future proofing (if the columns ever change?)
___
*[`pkg/workload/schemachange/schemachange.go` line 350 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-Ocg3DEb2JnNoUTeF3v2:-Ocg3DEb2JnNoUTeF3v3:b-eu3nad) ([raw file](https://github.com/cockroachdb/cockroach/blob/fafc7438b6b7a6969bc904f995fc63a1d87828f6/pkg/workload/schemachange/schemachange.go#L350)):*
> ```Go
> 			shutdownErr := tracerProvider.Shutdown(ctx)
> 			s.schemaWorkloadResultAnnotator.logWorkloadStats(stdoutLog)
> 			return errors.CombineErrors(inspectErr, errors.CombineErrors(closeErr, shutdownErr))
> ```

Lets use errors.Join instead.
___
*[`pkg/workload/schemachange/operation_generator.go` line 3460 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-Ocg2yBj2FTA_RQALBGy:-Ocg2yBj2FTA_RQALBGz:bxhkvwu) ([raw file](https://github.com/cockroachdb/cockroach/blob/fafc7438b6b7a6969bc904f995fc63a1d87828f6/pkg/workload/schemachange/operation_generator.go#L3460)):*
> ```Go
> 
> func (og *operationGenerator) inspect(ctx context.Context, tx pgx.Tx) (*opStmt, error) {
> 	stmt := makeOpStmt(OpStmtDML)
> ```

This workload runs on mixed version, is it safe to create the job in a mixed version setting? Think we may want a version gate.


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156295)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi, @shailendra-patel, and @williamchoe3)

___
*[`pkg/workload/schemachange/operation_generator.go` line 3460 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-Ocg2yBj2FTA_RQALBGy:-OcgMZcA-enoKtebn0vF:bge7roq) ([raw file](https://github.com/cockroachdb/cockroach/blob/fafc7438b6b7a6969bc904f995fc63a1d87828f6/pkg/workload/schemachange/operation_generator.go#L3460)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

This workload runs on mixed version, is it safe to create the job in a mixed version setting? Think we may want a version gate.
</blockquote></details>

Good call. I updated optype.go for this.
___
*[`pkg/workload/schemachange/schemachange.go` line 350 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-Ocg3DEb2JnNoUTeF3v2:-OcgMj98Aae7IBcf2WWz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fafc7438b6b7a6969bc904f995fc63a1d87828f6/pkg/workload/schemachange/schemachange.go#L350)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Lets use errors.Join instead.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @fqazi reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156295#-:-OcgT_BQBgnflyiCCYua:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @shailendra-patel and @williamchoe3)



<!-- Sent from Reviewable.io -->

spilchen committed:
workload/schemachange: add INSPECT operation to random schema workload

Adds a new inspect operation to the schema change workload, enabling random
generation of INSPECT TABLE and INSPECT DATABASE statements.

Features:
- Support for TABLE/DB targets, AS OF SYSTEM TIME
- Always runs in DETACHED mode so that it can be run inside a transaciton
- Results checked post-run via SHOW INSPECT ERRORS

Errors reported in JSON, consistent with existing workload logs

Closes #155483
Epic: CRDB-55075
Release note: none

spilchen commented:
TFTR!

bors r+
